### PR TITLE
optimization and refactoring the verify function

### PR DESCRIPTION
**Pull request by tuxuuman**
### What does it do?

Added caching of current user's policies into auth object.

### Why is it needed?

Allows you not to make many queries to the database when looking for data with related entities. Which speeds up the processing of requests several times

### How to test it?

Create a large collection of data with related entities and try a search query with a population of all entities.

Compare the request processing speed with the old version

### Related issue(s)/PR(s)

I do not know

### CODE REVIEW

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
This PR removes some of the important conditions in the verify. 

The idea is good but we need to still have the same logic as before :)

File: packages/plugins/users-permissions/server/strategies/users-permissions.js [hunk 1]
```diff
@@ -67,53 +67,26 @@ const authenticate = async ctx => {
...
-  const allowedActions = map('action', permissions);
+  if (!allowedActions) {
+    const permissions = await strapi.query('plugin::users-permissions.permission').findMany({
+      where: { role: user ? user.role.id : { type: 'public' } },
+    });
 
-  // An authenticated user can access non scoped routes
-  if (!config.scope) {
-    return;
+    allowedActions = auth.allowedActions = map('action', permissions);
```
Comment (alexandrebodin) [inline packages/plugins/users-permissions/server/strategies/users-permissions.js:82]: can you split this into multiple assignments instead

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

File: packages/plugins/users-permissions/server/strategies/users-permissions.js (hunk 1 above)
Comment (tuxuuman) [reply packages/plugins/users-permissions/server/strategies/users-permissions.js:82]: I did not quite understand what you want, but I noticed that I missed one more important condition

Comment (alexandrebodin) [reply packages/plugins/users-permissions/server/strategies/users-permissions.js:87]: can you avoid using = twice on the same line but do two assignements :)

File: packages/plugins/users-permissions/server/strategies/users-permissions.js [hunk 2]
```diff
@@ -67,53 +67,31 @@ const authenticate = async ctx => {
...
```
Comment (alexandrebodin) [inline packages/plugins/users-permissions/server/strategies/users-permissions.js:87]:
```suggestion
    allowedActions =  map('action', permissions);
    auth.allowedActions = allowedActions
```

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

**Review by alexandrebodin — APPROVED** [review]
LGTM

**Pull request merged**